File: Concurrent_Toy_Language_Interpreter/ADTs/MyList.java
package ADTs;

import Exceptions.EmptyContainerException;

import java.util.LinkedList;

public class MyList<T> implements IList<T> {
    private LinkedList<T> list;

    public MyList() {
        this.list = new LinkedList<>();
    }

    public void addFirst(T elem) {
        this.list.addFirst(elem);
    }

    public boolean isEmpty() {
        return this.list.isEmpty();
    }

    public T remove() throws EmptyContainerException {
        if(this.isEmpty())
            throw new EmptyContainerException();
        return this.list.remove();
    }

    public LinkedList<T> getList() {
        return this.list;
    }

    public String toString() {
        StringBuilder result = new StringBuilder();
        int size = 100;
        result.setLength(size);
        for(T elem : this.list) {
            if(result.length() > size) {
                size = 2 * size;
                result.setLength(size);
            }
            result.append(elem.toString());
            result.append("\n");
        }
        return result.toString();
    }
}

File: Toy_Language_Interpreter_C/ADTs/MyList.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace a7
{
    class MyList<T> : IList<T>
    {
        private LinkedList<T> list;
        public LinkedList<T> List
        {
            get
            {
                return this.list;
            }
        }

        public MyList()
        {
            this.list = new LinkedList<T>();
        }

        public void AddFirst(T elem)
        {
            this.list.AddFirst(elem);
        }

        public T Remove()
        {
            T node = this.list.ElementAt(0);
            this.list.RemoveLast();
            return node;
        }

        public bool IsEmpty()
        {
            if (this.list.ToArray().Length == 0)
                return true;
            return false;
        }

        public override String ToString()
        {
            String result = "";
            T[] temp = this.list.ToArray();
            foreach(T elem in temp)
            {
                result = result + elem.ToString() + "\n";
            }
            return result;
        }
    }
}
